### ISSUE TITLE
Dota runs out of memory when on different VT/xserver during Watch/Live preview

### ISSUE THREAD

**Issue by rgehlen** [ValveSoftware/Dota-2#1035]
How to reproduce:

Run dota2
Go to Watch -> Live
Switch to VT or different xserver
Use your favourite tool to watch dota memory usage go up indefinitely
Memory will keep getting allocated but never freed, until you switch back. (Some kind of garbage collector not running?)
If you wait long enough, see oom-killer in action.

Note that this is on Arch Linux with Skylake HD530 gpu and some recent mesa stuff. Not sure if this reproduces anywhere else.

**Comment by gdrewb-valve** [comment]
This only happens if you switch away from Dota?  If you don't switch the memory usage does not explode?

**Comment by rgehlen** [comment]
Yes, its not a critical bug whatsoever.

**Comment by gdrewb-valve** [comment]
If you let Dota accumulate memory but switch back to Dota before it gets killed does the memory usage stabilize or drop back down?  I'm just wondering if it's something building up waiting for Dota to be foreground.

**Comment by rgehlen** [comment]
Yes. I assumed "Memory will keep getting allocated but never freed, until you switch back." was a pretty decent description.

**Comment by gdrewb-valve** [comment]
OK, thanks.

**Comment by danginsburg** [comment]
I have been able to reproduce this on NVIDIA when doing a VT switch.  I believe what is happening is that the persistently mapped buffers that we have are becoming invalid without notification and then we crash.  I'm not sure if the issue on Intel is exactly the same.  There is likely not much we will do about this.  NVIDIA has an extension NV_robustness_video_memory_purge but I'm not sure it's worthwhile to implement and it won't help on Skylake.
